### PR TITLE
Implement sortedIndexOf

### PR DESCRIPTION
**Pull request by kriskowal**
r @raynos @jcorbin

### CODE REVIEW

File: lib/sorted-index-of.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+// Adapted from http://jsperf.com/binarysearch-vs-indexof/4
+
+var DEFAULT_THRESHOLD = 11;
+
+function sortedIndexOf(array, value, threshold) {
+    var min = 0;
+    var max = array.length - 1;
+    threshold = threshold || DEFAULT_THRESHOLD;
+
+    for (;;) {
```
Comment (jcorbin) [inline lib/sorted-index-of.js]: should be more of a "while not crossed over" loop usually

File: lib/sorted-index-of.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+        // Fall back to linear search for small ranges
+        if (min + threshold > max) {
```
Comment (jcorbin) [inline lib/sorted-index-of.js]: should be outside the bisection loop

**Comment by Raynos** [comment]
wat

**Comment by Raynos** [comment]
lgtm :P

**Comment by Raynos** [comment]
Wait; actually i didnt read it. I just assumed it was correct XD

**Comment by jcorbin** [comment]
:+1:

**Pull request merged**